agogosml/agogosml/writer/output_writer.py

```diff
@@ -1,40 +1,32 @@
-# -*- coding: utf-8 -*-
-""" OutputWriter class"""
+from typing import Callable
+from typing import Optional
 
 from agogosml.common.abstract_streaming_client import AbstractStreamingClient
 from agogosml.common.listener_client import ListenerClient
 from agogosml.utils.logger import Logger
 
 logger = Logger()
 
+Callback = Callable[[str], bool]
```

Comment:
Why [str], bool?

Reply:
The first bracketed list lists the arguments, everything after the comma is the return type. So this says that Callback is a function from string to boolean. See also here: https://docs.python.org/3/library/typing.html#typing.Callable